lib/SIL/Verifier/ReborrowVerifier.cpp

```diff
@@ -0,0 +1,63 @@
+//===--- ReborrowVerifier.cpp ----------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+
+#define DEBUG_TYPE "sil-reborrow-checker"
+
+#include "ReborrowVerifierPrivate.h"
+
+using namespace swift;
+
+bool ReborrowVerifier::checkGuaranteedPhiArgWithoutLifetimeEndingUses(
+    SILPhiArgument *phiArg) {
+  assert(phiArg->getOwnershipKind() == ValueOwnershipKind::Guaranteed);
+  if (!getRequiresLifetimeEndUse(phiArg))
+    return false;
+
+  errorBuilder.handleMalformedSIL([&] {
+    llvm::errs() << "Non trivial values, non address values, and non "
+                    "guaranteed function args must have at least one "
+                    "lifetime ending use?!\n";
+    llvm::errs() << "Value: " << *phiArg << '\n';
+  });
+  return true;
+}
+
+bool ReborrowVerifier::verify() {
+  bool result = false;
+  // For all guaranteed phi args and base value pairs in the worklist, verify
+  // that the guaranteed phi arg lies within the lifetime of the base value.
+  while (!worklist.empty()) {
+    auto item = worklist.pop_back_val();
+    auto *phiArg = item.first;
+    auto baseVal = item.second;
+    SmallVector<Operand *, 4> phiArgUses(phiArg->getUses());
```

Comment:
I would hoist these small things out. That way if we go big, we do not need to reallocate.